tests/spatial/fss_test_data.py

```diff
@@ -0,0 +1,74 @@
+"""
+Test data for Forecast Skill Score (FSS)    
+"""
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+
+def generate(obs_pdf, fcst_pdf, *, seed=42):
+    """
+    Generates random 2-D array for obs/fcst representing a 2-D grid
+    Args:
+        obs_pdf: (mu, stddev)
+        fcst_pdf: (mu, stddev)
+
+    where,
+        mu := mean
+        stddev := standard deviation
+    in,
+        Normal distribution ~ N(mu, stddev)
+    """
+    np.random.seed(seed)
+    h = 40
+    w = 60
+    obs = np.random.normal(loc=obs_pdf[0], scale=obs_pdf[1], size=(h, w))
```

Comment:
Will do

Reply:
@nicholasloveday thanks for prompting this test.

With `np.mesh` due to indexing reasons, the implementation will end up using a off-by-one window size in the first row/column. This may have also been an issue in the reference implementation, but the zero-padding would have "softened" its impact.

I didn't catch this in the `numba`/`rust` implementations because I used for loops and it was fine.

Surprisingly, this wouldn't have been caught in about 50% of the random sampling tests even though they were using relatively small `60x40 dataset with 10x10 windows`. Regardless, this shows the importance of curated tests in these scores which I chose to be `3x3` windows on a `5x4` matrix, that I could verify with pen and paper.

The fix is relatively simple - when using meshgrid for indexing, we simply insert an extra row/column of zeros before the first row/column - after which we can perform the integral image calculations as per usual. Note: this is different to the "zero-padding" option. Further, because we're using the clipping method for the zero-padding (as per the reference paper - which is somewhat of a shortcut), this needs to be done for that as well.

Reply:
Ok I added a suite of manual testing, and it all seems fine now, front insertion of a single row/column of zeros seems to have worked.

I think there may have been a slight inaccuracy in the implementation from the initial reference. Regardless, the existing test results that was more true to the reference implementation didn't change by much after this change: `<0.001`. While, edge cases in small windows are now handled properly.